.github/workflows/build_test_validate.yml

```diff
@@ -0,0 +1,30 @@
+
+name: Build and test all the Kiota packages
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        environment:
+          name: build_test
+        strategy:
+          matrix:
+            node-version: [14.x, 16.x]
+        steps:
+        - uses: actions/checkout@v3
+        - name: Use Node.js ${{ matrix.node-version }}
+          uses: actions/setup-node@v3
+          with:
+            node-version: ${{ matrix.node-version }}
+        - run: yarn install --frozen-lockfile
+        - run: yarn run build
+        - run: lerna run test:integrated
+        - run: lerna run test --parallel
```

Comment:
isn't this missing a deploy job?

Reply:
Versioning and deploy job will be a separate workflow.

Reply:
what are you planning to be the trigger of releases? a tag creation? push on main? something else?

Reply:
The plan is to use lerna version which updates the package version as well as updates the dependencies and creates the tags. 
This way the dependent packages are in sync. If the package versions are not in sync, then a dependent packages looks for the parent package in npm instead of using the local code. With manual version updates, it is easy to miss updating version numbers in every place. 

The versioning part will be automated in #93. 

The process is to run:
`lerna version`  plus a customized automated process followed by
`lerna publish` publishes to npm.

Reply:
so if I understand correctly, between the time we merge this PR and the time we merge the PR for #93, we won't be able to release?

Reply:
We will be able to release. I will manually update the versions using lerna commands and publish off those tags. #93 includes some work to update version with custom suffixes. 

The main objective of this PR is too have the initial set up and enable easy testing and validation to include further PRs. 

publish and versioning wont be blocked with these changes.

Reply:
I might be missing something, but once you push the new tags, what's going to perform the npm publish?

Reply:
resolving as this was just added, thanks!